datajunction-server/datajunction_server/construction/build.py

```diff
@@ -281,12 +286,18 @@ def _build_tables_on_select(
         )  # got a materialization
         if node_table is None:  # no materialization - recurse to node first
             node_query = parse(cast(str, node.query))
-            node_table = build_ast(  # type: ignore
-                session,
-                node_query,
-                build_criteria,
-            ).select  # pylint: disable=W0212
-            node_table.parenthesized = True  # type: ignore
+            if hash(node_query) in memoized_queries:  # pragma: no cover
+                node_table = memoized_queries[hash(node_query)].select  # type: ignore
+            else:
+                query_ast = build_ast(  # type: ignore
+                    session,
+                    node_query,
+                    memoized_queries,
+                    build_criteria,
+                )
+                node_table = query_ast.select  # type: ignore
+                node_table.parenthesized = True  # type: ignore
```

Comment:
What does this mean?

Reply:
This is a flag that lets the renderer know if this table object needs to be surrounded by parentheses. Which is the case when a from table is a subquery.